manifests/certificate/x509.pp

```diff
@@ -193,15 +193,16 @@
     encrypted   => $encrypted,
   }
   ~> x509_cert { $_crt:
-    ensure   => $ensure,
-    template => $_cnf,
-    csr      => $_csr,
-    days     => $days,
-    password => $password,
-    req_ext  => $req_ext,
-    force    => $force,
-    ca       => $ca,
-    cakey    => $cakey,
+    ensure      => $ensure,
+    template    => $_cnf,
+    csr         => $_csr,
+    private_key => $_key,
```

Comment:
This isn't part of the current code. It happens to work now because `private_key` defaults to the same value as `$_key` defaults to. If a user passes in `$key` it will fail, so this is still needed.

Reply:
https://github.com/voxpupuli/puppet-openssl/pull/186 only does this part. If that's merged, we can close this PR.